WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift

```diff
@@ -596,3 +597,243 @@ extension ReaderPostCardCell: ReaderCardDiscoverAttributionViewDelegate {
+        let authorName = postAuthor()
+        let blogTitle = blogName()
+
+        return headerButtonAccessibilityLabel(name: authorName, title: blogTitle) + "," + postTitle() + "," + postContent()
+    }
+
+    private func cardAccessibilityHint() -> String {
+        return NSLocalizedString("Shows the post content", comment: "Accessibility hint for the Reader Cell")
+    }
+
+    private func prepareHeaderButtonForVoiceOver() {
+        guard headerBlogButtonIsEnabled else {
+            /// When the headerbutton is disabled, hide it from VoiceOver as well.
+            headerBlogButton.isAccessibilityElement = false
+            return
+        }
+
+        headerBlogButton.isAccessibilityElement = true
+
+        let authorName = postAuthor()
+        let blogTitle = blogName()
+
+        headerBlogButton.accessibilityLabel = headerButtonAccessibilityLabel(name: authorName, title: blogTitle)
+        headerBlogButton.accessibilityHint = headerButtonAccessibilityHint(title: blogTitle)
+        headerBlogButton.accessibilityTraits = UIAccessibilityTraitButton
+    }
+
+    private func headerButtonAccessibilityLabel(name: String, title: String) -> String {
+        return authorNameAndBlogTitle(name: name, title: title) + "," + datePublished()
+    }
+
+    private func authorNameAndBlogTitle(name: String, title: String) -> String {
+        let format = NSLocalizedString("Post by %@, from %@", comment: "Spoken accessibility label for blog author and name in Reader cell.")
+
+        return String(format: format, name, title)
+    }
+
+    private func headerButtonAccessibilityHint(title: String) -> String {
+        let format = NSLocalizedString("Shows all posts from %@", comment: "Spoken accessibility hint for blog name in Reader cell.")
+        return String(format: format, title)
+    }
+
+    private func prepareShareForVoiceOver() {
+        shareButton.accessibilityLabel = NSLocalizedString("Share", comment: "Spoken accessibility label")
+        shareButton.accessibilityHint = NSLocalizedString("Shares this post", comment: "Spoken accessibility hint for Share buttons")
+        shareButton.accessibilityTraits = UIAccessibilityTraitButton
+    }
+
+    private func prepareCommentsForVoiceOver() {
+        commentActionButton.accessibilityLabel = commentsLabel()
+        commentActionButton.accessibilityHint = NSLocalizedString("Shows comments", comment: "Spoken accessibility hint for Comments buttons")
+        commentActionButton.accessibilityTraits = UIAccessibilityTraitButton
+    }
+
+    private func commentsLabel() -> String {
+        let commentCount = contentProvider?.commentCount()?.intValue ?? 0
+
+        let format = commentCount > 1 ? pluralCommentFormat() : singularCommentFormat()
+
+        return String(format: format, "\(commentCount)")
+    }
+
+    private func singularCommentFormat() -> String {
+        return NSLocalizedString("%@ comment", comment: "Accesibility label for comments button (singular)")
```

Comment:
Nitpick, typo: `Accesibility` -> `Accessibility`. Same in the plural method below.